src/auth/auth.ts

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 Artificial, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import * as vscode from 'vscode';
+import { addFileToContext } from '../utils';
+import { OutputLog } from '../providers/outputLogProvider';
+
+export async function authExternalUriRegistration(context: vscode.ExtensionContext) {
+    // Register a URI handler for the authentication callback
+  vscode.window.registerUriHandler({
+    handleUri(uri: vscode.Uri): vscode.ProviderResult<void> {
+      if (uri.path === '/auth-complete') {
+        // Parse query parameters
+        const query = new URLSearchParams(uri.query);
+        const instanceURL = query.get('instanceURL');
+        const token = query.get('token');
+
+        if (instanceURL === null || token === null) {
+          vscode.window.showErrorMessage('Sign in failed: missing instanceURL or token');
+          return;
+        }
+        // Convert instanceURL into a valid host for AC
+        const u = new URL(instanceURL);
+
+        const generatedObj = {
+          artificial: {
```

Comment:
So we'll just need to set the lab id in the main config file?

Reply:
Yes. As a future request, we could have a lab picker in the ui that opens in alab